### ISSUE TITLE
Ensure Each HTML Document Has One Main Landmark and Contains All Page Content Within Landmarks

### ISSUE THREAD

**Issue by ipreencekmr** [expressjs/expressjs.com#1807]
To ensure proper navigation and accessibility, it is best practice to use both HTML5 and ARIA landmarks to structure page content. All content should be contained within appropriate landmark elements to provide clear navigation regions. In HTML5, elements such as `<header>`, `<nav>`, `<main>`, and `<footer>` should be used. Their ARIA counterparts are _role="banner"_, _role="navigation"_, _role="main"_, and _role="contentinfo"_, respectively.

<img width="953" alt="Image" src="https://github.com/user-attachments/assets/b97acfa0-4115-49e1-ad46-ea2faa9c2a9e" />

**Reference Links** 

- https://dequeuniversity.com/rules/axe/4.10/landmark-one-main?application=AxeChrome
- https://dequeuniversity.com/rules/axe/4.10/region?application=AxeChrome

Impact: **Moderate**

**Benefits:**

- Enhances screen reader navigation, allowing users with visual impairments to move efficiently between sections.
- Improves keyboard accessibility, enabling users to jump between content regions without excessive tabbing.
- Boosts SEO and content discoverability, as search engines better understand the page structure.
- Ensures compliance with WCAG and accessibility laws, reducing legal risks and improving inclusivity.

**Page Affected:**

_**Getting Started**_

1. https://expressjs.com/en/starter/installing.html
2. https://expressjs.com/en/starter/hello-world.html
3. https://expressjs.com/en/starter/generator.html
4. https://expressjs.com/en/starter/basic-routing.html
5. https://expressjs.com/en/starter/static-files.html
6. https://expressjs.com/en/starter/examples.html
7. https://expressjs.com/en/starter/faq.html

_**Guide**_

1. https://expressjs.com/en/guide/routing.html
2. https://expressjs.com/en/guide/writing-middleware.html
3. https://expressjs.com/en/guide/using-middleware.html
4. https://expressjs.com/en/guide/overriding-express-api.html
5. https://expressjs.com/en/guide/using-template-engines.html
6. https://expressjs.com/en/guide/error-handling.html
7. https://expressjs.com/en/guide/debugging.html
8. https://expressjs.com/en/guide/behind-proxies.html
9. https://expressjs.com/en/guide/migrating-4.html
10. https://expressjs.com/en/guide/migrating-5.html
11. https://expressjs.com/en/guide/database-integration.html

_**API Reference**_
_**Advanced Topics**_
_**Resources**_
_**Support**_
_**Blog**_

**Suggested Fix**

- Include body content inside main html-5 element except header and footer 

- <img width="310" alt="Image" src="https://github.com/user-attachments/assets/b32bfcba-b1f4-43a4-80d9-8f0a0a03121f" />

- <img width="203" alt="Image" src="https://github.com/user-attachments/assets/2f525733-f200-4319-ab0f-ddcd50bb14fb" />

**An Ideal page**

```
<header role="banner">
   <p>Put company logo, etc. here.</p>
</header>
<nav role="navigation">
   <ul>
      <li>Put navigation here</li>
   </ul>
</nav>
<main role="main">
   <p>Put main content here.</p>
</main>
<footer role="contentinfo">
   <p>Put copyright, etc. here.</p>
</footer>
```

**Testing Tool**

[Axe DevTools](https://chromewebstore.google.com/detail/lhdoppojpmngadmnindnejefpokejbdd?utm_source=item-share-cp) 

**Apply Fix Menu Wise** 

We can divide PR by Menu sequentially and create branch name with a standard format to keep consistency across branch name. 

**Suggested Branch Name Format** 

**Format:** `defect-a11y-{ISSUE_NO}-{MENU}`
**Example:** `defect-a11y-1807-getting-started`

**Comment by bjohansebas** [comment]
this issue is more complicated than it seems. I would prefer to work on most of this issue together with https://github.com/expressjs/expressjs.com/issues/1780, as they are quite related in the way they will be resolved

**Comment by ShubhamOulkar** [comment]
@ipreencekmr you can fix docs in #1775, My plan was to fix it last. In this PR layout is changed for sidebar menus. So it is better to fix accessibility there. 

I think header needs redesign as it is referred in #1780. 
Header has more accessibility issues like theme and language btn's are not accessible by keyboard and many more. It is better to open pr to just fix accessibility issues and redesign header. Same goes with footer.
You can use excel format referenced in #1753.

**Comment by ipreencekmr** [comment]
> [@ipreencekmr](https://github.com/ipreencekmr) you can fix docs in [#1775](https://github.com/expressjs/expressjs.com/pull/1775), My plan was to fix it last. In this PR layout is changed for sidebar menus. So it is better to fix accessibility there.
> 
> I think header needs redesign as it is referred in [#1780](https://github.com/expressjs/expressjs.com/issues/1780). Header has more accessibility issues like theme and language btn's are not accessible by keyboard and many more. It is better to open pr to just fix accessibility issues and redesign header. Same goes with footer. You can use excel format referenced in [#1753](https://github.com/expressjs/expressjs.com/issues/1753).

@ShubhamOulkar I have updated excel sheet with issues we had 

https://docs.google.com/spreadsheets/d/1SWpouB0Nggze09lrCtLcTmyFvnYZOvNTIRAg9bisoYs/edit?gid=1766402489#gid=1766402489

I would like to have more information on how header should be designed, do we have any design in mind ?

**Comment by ShubhamOulkar** [comment]
Hey @ipreencekmr, you've updated in the wrong tab. The appendix accessibility guidelines, defined by W3C, must be followed by developers to ensure web accessibility. I've listed accessibility issues under the Success Criteria column in the 'Issues Found' tab. You have to fill the sheet using data from the 'Issues Found' tab. Please read the template README. [See for more information](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA11.html)

**Comment by ipreencekmr** [comment]
> README

Hi @ShubhamOulkar, I will list out issues in Issues Found tab now on if I find any. Are we expecting any values in drop down of the `Issues Found` tab columns. 

![Image](https://github.com/user-attachments/assets/7fbe2153-d348-43c1-b7d0-a09addcecd2f)

**Comment by ShubhamOulkar** [comment]
### Latest update 
Landmark content issue is resolved in #1775. Only navigation header need landmark fix.

**Comment by ShubhamOulkar** [comment]
Hi @ipreencekmr, I reviewed this issue. Adding role=navigation on header as suggested by tool is not correct. header element has implicit aria role [banner](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/banner_role). I verified in 👇🏻 which is correct it should be banner. See banner link for more information.

![Image](https://github.com/user-attachments/assets/672dea63-644e-4241-881e-0b9733869478)

But on further investigation there is role=navigation a11y issue as show in following 👇🏻, `<div id="navbar"></div>` should be `<nav></nav>` element. May be AXE dev tool failed to suggest correct fix.

![Image](https://github.com/user-attachments/assets/bcfefdea-5528-41fd-a6fc-127373bafdc0)

I suggest following fix 👇🏻

![Image](https://github.com/user-attachments/assets/e0d57389-49ff-432c-b078-f910c3b9cfd6)

After fix :- 

![Image](https://github.com/user-attachments/assets/55717539-af23-460f-8449-cb927f4c9681)
